File: javascript/leetcode/405. Convert a Number to Hexadecimal.js
/**
 * @param {number} num
 * @return {string}
 */
var toHex = function (num) {
  const _hex = "0123456789abcdef";

  if (!num) return "0";
  if (num < 0) num = num >>> 0;

  let ans = [];

  while (num) {
    ans.push(_hex[num % 16]);
    num = parseInt(num / 16);
  }

  return ans.reverse().join("");
};

File: c++/leetcode/405. Convert a Number to Hexadecimal.cpp
class Solution
{
public:
  string toHex(int num)
  {
    const string _hex = "0123456789abcdef";

    if (!num)
      return "0";
    unsigned n = num;

    string ans{};

    while (n)
    {
      ans.push_back(_hex[n % 16]);
      n /= 16;
    }

    reverse(ans.begin(), ans.end());
    return ans;
  }
};